oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/Optimizer.kt

```diff
@@ -0,0 +1,18 @@
+package com.empowerops.volition.ref_oasis
+
+import kotlin.random.Random
+
+interface Optimizer {
+    fun generateInputs(inputs: List<Input>): Map<String, Double>
+}
+
+class RandomNumberOptimizer : Optimizer {
```

Comment:
you could really just leave this as `object Optimizer {`

I dont think there's a good reason to have an interface here.

Reply:
I think this going back to thing we were talking about, I have plan to have deterministic optimizer electable and that's why I come up with this interface.